Tasks/UseDotNetV1/usedotnet.ts

```diff
@@ -0,0 +1,324 @@
+    }
+    const accessToken: string = pkgLocationUtils.getSystemAccessToken();
+    const feedUri = await nutil.getNuGetFeedRegistryUrl(packagingLocation.DefaultPackagingUri, feedName, null, accessToken, true);
+
+    addNugetFeed(feedName, feedUri, 'VSTS');
+}
+
+async function addExternalFeed(feedName: string) {
+    const externalAuthArr = commandHelper.GetExternalAuthInfoArray('externalEndpoint');
+    const externalAuth = externalAuthArr[0];
+
+    if (!externalAuth) {
+        tl.setResult(tl.TaskResult.Failed, tl.loc('Error_NoSourceSpecifiedForPush'));
+        return;
+    }
+
+    const feedUri = externalAuth.packageSource.feedUri;
+
+    const authType: auth.ExternalAuthType = externalAuth.authType;
+    let apiKey = '';
+    let additionalArgs: Array<string> = [];
+    switch (authType) {
+        case (auth.ExternalAuthType.UsernamePassword):
+            const usernameInfo = externalAuth as auth.UsernamePasswordExternalAuthInfo;
+            additionalArgs = ['-UserName', usernameInfo.username, '-Password', usernameInfo.password];
+            break;
+        case (auth.ExternalAuthType.Token):
+            apiKey = 'RequiredApiKey';
+            break;
+        case (auth.ExternalAuthType.ApiKey):
+            const apiKeyAuthInfo = externalAuth as auth.ApiKeyExternalAuthInfo;
+            apiKey = apiKeyAuthInfo.apiKey;
+            break;
+        default:
+            break;
+    }
+
+    addNugetFeed(feedName, feedUri, apiKey, additionalArgs);
+}
+
+function addNugetFeed(feedName: string, feedUri: string, apiKey: string = '', additionalArgs: Array<string> = []) {
+    const nugetPath = tl.which('nuget');
+
+    // Add feed
+    let nuget = tl.tool(nugetPath);
+    nuget.arg('sources');
+    nuget.arg('Add');
+    
+    nuget.arg('-Name');
+    nuget.arg(feedName);
+    
+    nuget.arg('-Source');
+    nuget.arg(feedUri);
+
+    additionalArgs.forEach(arg => {
+        nuget.arg(arg);
+    });
+
+    nuget.arg('-NonInteractive');
+
+    nuget.exec({} as trm.IExecOptions);
+
+    if (apiKey) {
+        nuget = tl.tool(nugetPath);
+        nuget.arg('setapikey');
+        nuget.arg(apiKey);
+
+        nuget.arg('-Source');
+        nuget.arg(feedUri);
+
+        nuget.arg('-NonInteractive');
+
+        nuget.exec({} as trm.IExecOptions);
+    }
+}
+
+async function run() {
+    // set the console code page to "UTF-8"
+    if (tl.osType() === 'Windows_NT') {
+        try {
+            tl.execSync(path.resolve(process.env.windir, "system32", "chcp.com"), ["65001"]);
+        }
+        catch (ex) {
+            tl.warning(tl.loc("CouldNotSetCodePaging", JSON.stringify(ex)));
+        }
+    }
+
+    let packageType = tl.getInput('packageType') || 'sdk';
+    const version: string = tl.getInput('version');
+    if (version) {
+        console.log(tl.loc("ToolToInstall", packageType, version));
+        await new DotnetCoreInstaller(packageType, version).install();
+    }
+
+    const proxy: boolean = tl.getBoolInput('proxy');
```

Comment:
Assertion: This shouldn't be an input.  If proxy is configured, use it.  We don't want proxy bool on every task (logical conclusion).  We have proxy exclusion rules for scenarios where it's not configured and that's based more on the url rather than a bool input